py/server/deephaven/barrage.py

```diff
@@ -0,0 +1,180 @@
+    return BarrageSession(j_barrage_session_factory.newBarrageSession())
+
+
+def _get_barrage_session_direct(client_config, auth) -> BarrageSession:
+    """Note, this is used for testing only. This way of constructing a Barrage session is less efficient because it does
+    not share any of the state or configuration that the server provides; namely, when you are doing it with the server
+    context it provides a singleton executor, allocator, outbound SSL configuration, and the ability for the server to
+    hook in additional channel building options.
+    """
+    j_channel = _JChannelHelper.channel(client_config)
+    j_dh_channel = _JDeephavenChannelImpl(j_channel)
+
+    j_session_config = (_JSessionImplConfig.builder()
+                        .executor(_JExecutors.newScheduledThreadPool(4))
+                        .authenticationTypeAndValue(auth)
+                        .channel(j_dh_channel)
+                        .build())
+    j_session = _JSessionImpl.create(j_session_config)
+    return BarrageSession(_JBarrageSession.create(j_session, _JRootAllocator(), j_channel))
+
+
+def _build_client_config(target_uri, tls_root_certs) -> jpy.JType:
+    j_client_config_builder = _JClientConfig.builder()
+    j_client_config_builder.target(_JDeephavenTarget.of(_JURI(target_uri)))
+    if tls_root_certs:
+        j_ssl_config = _JSSLConfig.builder().trust(
+            _JTrustCustom.ofX509(tls_root_certs, 0, len(tls_root_certs))).build()
+        j_client_config_builder.ssl(j_ssl_config)
+    j_client_config = j_client_config_builder.build()
+    return j_client_config
+
+
+class BarrageSession(JObjectWrapper):
+    """ A Deephaven Barrage session to a remote server."""
+
+    j_object_type = _JBarrageSession
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_barrage_session
+
+    def __init__(self, j_barrage_session):
+        self.j_barrage_session = j_barrage_session
+        self.j_session = j_barrage_session.session()
+
+    def subscribe(self, ticket: bytes) -> Table:
+        """ Subscribes to a published remote table with the given ticket.
+
+        Note, if the remote table is closed or its owner session is closed, the ticket becomes invalid.
+
+        Args:
+            ticket (bytes): the bytes of the ticket
+
+        Returns:
+            a Table
+
+        Raises:
+            DHError
+        """
+        try:
+            j_table_handle = self.j_session.of(_JTableSpec.ticket(ticket))
+            j_barrage_subscription = self.j_barrage_session.subscribe(j_table_handle,
+                                                                      _JBarrageTableResolver.SUB_OPTIONS)
+            return Table(j_barrage_subscription.entireTable().get())
+        except Exception as e:
+            raise DHError(e, "failed to subscribe to the remote table with the provided ticket.") from e
+
+    def snapshot(self, ticket: bytes) -> Table:
```

Comment:
Same question here.  Anything else we may want to snapshot?  If so, maybe `snapshot_table`.